src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -4147,15 +4146,23 @@ void ExportMusicXml::chord(Chord* chord, staff_idx_t staff, const std::vector<Ly
 
         m_xml.tag("voice", static_cast<int>(voice));
 
-        writeTypeAndDots(m_xml, note);
+        writeType(m_xml, note);
+        for (NoteDot* dot : note->dots()) {
+            String dotTag = u"dot";
+            dotTag += color2xml(dot);
+            dotTag += elementPosition(this, dot);
+            m_xml.tagRaw(dotTag);
+        }
         writeAccidental(m_xml, u"accidental", note->accidental());
         writeTimeModification(m_xml, note->chord()->tuplet(), tremoloCorrection(note));
 
         // no stem for whole notes and beyond
         if (chord->noStem() || chord->measure()->stemless(chord->staffIdx())) {
             m_xml.tag("stem", "none");
         } else if (note->chord()->stem()) {
-            m_xml.tag("stem", note->chord()->up() ? "up" : "down");
+            String stemTag = u"stem";
+            stemTag += color2xml(toStem(note->chord()->stem()));
```

Comment:
Perhaps:
```suggestion
        } else if (const Stem* = note->chord()->stem()) {
            String stemTag = u"stem";
            stemTag += color2xml(stem);
```
(removing redundant `toStem`)

Reply:
Changed and extended the color export test a bit.